pkg/config/kpconfig.go

```diff
@@ -1,8 +1,159 @@
 package config
 
-import corev1 "k8s.io/api/core/v1"
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/vmware-tanzu/kpack-cli/pkg/k8s"
+)
+
+const (
+	kpNamespace                         = "kpack"
+	kpConfigMapName                     = "kp-config"
+	canonicalRepositoryKey              = "canonical.repository"
+	canonicalServiceAccountNameKey      = "canonical.repository.serviceaccount"
+	canonicalServiceAccountNamespaceKey = "canonical.repository.serviceaccount.namespace"
+)
 
 type KpConfig struct {
-	CanonicalRepository string
-	ServiceAccount      corev1.ObjectReference
+	canonicalRepository string
+	serviceAccount      corev1.ObjectReference
+}
+
+func NewKpConfig(canonicalRepository string, serviceAccount corev1.ObjectReference) KpConfig {
+	return KpConfig{
+		canonicalRepository: canonicalRepository,
+		serviceAccount:      serviceAccount,
+	}
+}
+
+func (c KpConfig) CanonicalRepository() (string, error) {
+	if c.canonicalRepository == "" {
+		return "", errors.Errorf("failed to get canonical repository: key %q not found in configmap %q", canonicalRepositoryKey, kpConfigMapName)
+	}
+
+	return c.canonicalRepository, nil
+}
+
+func (c KpConfig) ServiceAccount() (corev1.ObjectReference, error) {
+	if c.serviceAccount.Name == "" {
```

Comment:
Should this provide default if unset? 

I imagine the case could occur when the service account or repo is never set but, the user attempts to run `kp clusterbuilder create --tag` which does not need the repo?

Reply:
maybe we should release kpack with the `kp-config` set with just the canonical service account as `default`?

Reply:
that way we don't have to worry about defaulting the canonical service account in an unexpected way

Reply:
i think both options might be beneficial. Doing it in kp allows previous versions of kpack to get this functionality, while new versions will also get a kp-config to start with